compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BObjectType.java

```diff
@@ -56,4 +58,20 @@ public String toString() {
         return this.tsymbol.toString();
     }
 
+    @Override
+    public boolean hasImplicitInitialValue() {
+        if (this.tsymbol instanceof BObjectTypeSymbol) { // To please the FindBugs warning, BC_UNCONFIRMED_CAST
+            BAttachedFunction initializerFunc = ((BObjectTypeSymbol) this.tsymbol).initializerFunc;
+            if (initializerFunc == null) {
+                // No __init function found.
```

Comment:
An Object type can't have an implicit initial value if there any field with a type that doesn't have an implicit initial value. @SupunS pls confirm.

Reply:
As per spec, it's not allowed for a field which doesn't have a IIV to be uninitialized when __init exit.

Reply:
Yes, currently for public fields, it is mandatory to be initialized in the object __init(). However i'm not sure whether we do the same check for the private fields (we don't, IIRC).

If it is the case, i guess we need to update that as well.